.github/workflows/multi-arch-build.yaml

```diff
@@ -0,0 +1,141 @@
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+        with:
+          driver-opts: network=host
+          install: true
+
+      - name: Build and push local Buildah
+        uses: docker/build-push-action@v2
+        with:
+          context: contrib/buildahimage/${{ matrix.source }}
+          file: ./contrib/buildahimage/${{ matrix.source }}/Dockerfile
+          platforms: ${{ env.PLATFORMS }}
+          push: true
+          tags: |
+            localhost:5000/buildah/${{ matrix.source }}
+
+      # Simple verification that container works
+      - name: amd64 container sniff test
+        id: sniff_test
+        run: |
+          VERSION_OUTPUT="$(docker run localhost:5000/buildah/${{ matrix.source }} buildah version)"
+          echo "$VERSION_OUTPUT"
+          echo ::set-output name=version_output::$VERSION_OUTPUT
+
+      # Generate image related info - names, labels
+      - name: Generate image information
+        id: image_info
+        run: |
+          # have special environment variable for image name.
+          # Names are different for upstream and stable Buildah images
+          if [ "${{ matrix.source }}" == "upstream" ]; then
+            # quay.io/buildah/upstream:master
+            echo ::set-output name=buildah_tag::"${{ env.BUILDAH_QUAY_REGISTRY}}/${{ matrix.source }}:master"
+          fi
+
+          if [ "${{ matrix.source }}" == "stable" ]; then
+            VERSION=v"$(echo "${{ steps.sniff_test.outputs.version_output }}" | head -n 1 | awk '{print $2}')"
+            # quay.io/buildah/stable:vX.X.X
+            echo ::set-output name=buildah_tag::"${{ env.BUILDAH_QUAY_REGISTRY}}/${{ matrix.source }}:${VERSION}"
+            # quay.io/contaners/buildah:vX.X.X
+            echo ::set-output name=containers_tag::"${{ env.CONTAINERS_QUAY_REGISTRY}}/buildah:${VERSION}"
+          fi
+
+          # have labels environment variable to use it in the further steps.
+          # Env variable is used to store multi-line value
+          echo "LABELS<<EOF" >> $GITHUB_ENV
+          echo "org.opencontainers.image.source=${{ github.event.repository.html_url }}" >> $GITHUB_ENV
+          echo "org.opencontainers.image.revision=${{ github.sha }}" >> $GITHUB_ENV
+          echo "org.opencontainers.image.created=$(date -u --iso-8601=seconds)" >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
+
+      # Separate steps to login and push for buildah and containers quay
+      # repositories are required, because 2 sets of credentials are used and `docker
+      # login` as well as `podman login` do not support having 2 different
+      # credential sets for 1 registry.
+      # At the same time reuse of non-shell steps is not supported by Github Actions via
+      # anchors or composite actions
+
+      # Push to buildah Quay repo stable and upstream Buildah
```

Comment:
do you handle testing too?  The testing branch is used for the latest test version in [bohdi](https://bodhi.fedoraproject.org/updates/?search=buildah).  Sometimes it's the same as stable, but oftentimes not.  Basically the latest beta version.

Reply:
@TomSweeneyRedHat so far it's only stable and upstream. However I think it's not a problem to add testing as third parallel build. Could you please list from which source to which image repo/tags you would like to see the builds? :)
So far I just replicated skopeo approach with push from stable to `quay.io/buildah/stable:v..` and `quay.io/containers/buildah:v..` and from upstream to `quay.io/buildah/upstream:master`

Reply:
Tom, Yulia's github-actions workflow is running a matrix across the different releases, but the source is always "latest master" (at the time of job triggering).  It's not actually differentiating between (conceptually) "stable" and "upstream" (or "testing") in any way.  It simply asks buildah itself for the version number, and tags image accordingly.  In other words, it's not updating what the "latest" tag points to, or behaving differently if it's triggered from a branch-push or a new tag-push.

Is there some machine-detectable buildah (or repo) detail which separates a *test* version from *stable/upstream*?

Reply:
oh, I thought the `testing` question was about location of Dockerfile - https://github.com/containers/buildah/tree/test/contrib/buildahimage/testing .. Sorry for misunderstanding :(

Reply:
No worries, most likely my bad stealing some late night time for stealth reviewing.  I was referring to the stable/upstream/testing repo's that we have for `quay.io/buildah/*`  and @cevich (as usual) straightened me out with his reply above.